Sources/StreamChatUI/Utils/ImageLoading/NukeImageLoader.swift

```diff
@@ -110,7 +109,8 @@ open class NukeImageLoader: ImageLoading {
         if resize && size != .zero {
             url = imageCDN.thumbnailURL(originalURL: url, preferredSize: size)
         }
-        
+        let cachingKey = imageCDN.cachingKey(forImage: url)
```

Comment:
At this point, the url contains the size parameters if size is not zero, and `resize` is true

Reply:
Can we double-check we provide non-nil `preferred size` and `resize: true` in all the places?

Reply:
I think something is weird here (not necessarily related to this change). What if I send an image with an image attachment that is not hosted on the CDN?

You can reproduce this using https://github.com/GetStream/sdk-tester#test-messages-with-attached-images if needed

Reply:
We are now passing `resize: true` everywhere.

But we are not passing `preferredSize` anywhere but where we have avatars. But in the case of not passing preferred size it would use the imageView.size, or it would store it using a caching key that has no parameters, and would not resize the image. Either way, we should be safe.

Where we are not passing preferredSize, is because we don't know its size (Name an uploaded image), and it should be kept like this. In most cases image is not resized because imageView.frame is zero.

Reply:
@tbarbugli in the case of not being a url that's coming from our CDN, it should be keeping the whole url as caching key. And thumbnailURL returns the same url when the host is not our CDN.

What I just spotted based on your comment is that the url will be kept the same, but the image can be resized 🤦 , which puts us in the same situation for non CDN originated urls 🦆

Reply:
@tbarbugli All images are loaded through this centralized function. So all images will be passed by the `imageCDN` AFAIK. 

@polqf This `resize: true` is very confusing. On the notion doc we should make sure we change this in the future to something like `aspectRatio` or `contentMode`. Or at least `clip: true` if on the backend we only support clipping.

Reply:
@nuno-vieira what we mean by CDN here is "the real CDN", so the CDN that's serving the images server side.

Also, `resize: true` does exactly what it says now, upon downloading the image, it will resize the image, and use that one. What it's not clear from my side is if the resized image is the cached one (which I'm guessing it's the case).

Reply:
one cause of confusion here is that we are mixing different responsibilities:

- uploading an image and get its URL in return
- retrieving an image via URL
- refreshing an expire URL 
- apply some processing to an image (ie. resizing)
- caching an image based on its URL
- caching the result of image processing (when its done on the device)

Reply:
Resolving this conversation for the sake of readability given that the implementation has been changed after the recent changes.

I completely agree with you @tbarbugli wrt the mixing of responsibilities. This is agreed to be taken care of in the upcoming work that is being debated in Notion (https://www.notion.so/stream-chat-fe-sdk/Image-Loading-375fcf50af6f4753964cb574e0d2bf6d). I'm adding your comment there for the sake of not losing all the findings.